### PR TITLE
Create gh-container-image.yml

### PR DESCRIPTION
**Pull request by isSerge**

### CODE REVIEW

File: .github/workflows/gh-container-image.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Update container image
+
+on:   
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (nazar-pc) [inline .github/workflows/gh-container-image.yml:9]: Why is this needed? It will build image on every PR against `main`. `push` above should be sufficient.

File: .github/workflows/gh-container-image.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME
+        working-directory: backend
+
+      - name: Log in to registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push image
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
```
Comment (nazar-pc) [inline .github/workflows/gh-container-image.yml:34]: This line seems redundant, we fully control the name here

File: .github/workflows/gh-container-image.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME
```
Comment (nazar-pc) [inline .github/workflows/gh-container-image.yml:25]: Why no build image with `ghcr.io/subspace/subspace-relayer:latest` tag right away though? You wouldn't need to re-tag it in that case.

Comment (liuchengxu) [reply .github/workflows/gh-container-image.yml:22]: If there are no other concerns, `ghcr.io/subspace/relayer:latest` would be more concise IMHO. That being said, we can have `subspace/node`, `subspace/farmer`, etc later.

Comment (isSerge) [reply .github/workflows/gh-container-image.yml:22]: I agree, however, we are not putting `node` and `farmer` on ghcr as discussed yesterday

Comment (liuchengxu) [reply .github/workflows/gh-container-image.yml:22]: Yeah, just said about the image naming, not saying whether they should be hosted on ghcr or dockerhub :P.

Comment (nazar-pc) [reply .github/workflows/gh-container-image.yml:22]: `ghcr.io/subspace/relayer:latest` is fine too, I'm just trying to keep everything consistent:
* package name
* directory name
* container image name

If all those things are absolutely identical, it makes life so much easier sometimes.

**Review by nazar-pc — DISMISSED** [review]
Looks fine, please squash on merge

**Review by nazar-pc — APPROVED** [review]
